app/controllers/entries_controller.rb

```diff
@@ -4,22 +4,32 @@ class EntriesController < ApplicationController
   DATE_FORMAT = "%d/%m/%Y".freeze
```

Comment:
I just see a lot of if/else statements which would suggest two seperate controllers would be better, you could also do it with inheritance, put methods like `resource` in the parent controller and create method in the child controller.